controllers/serviceclaim_controller.go

```diff
@@ -183,28 +183,70 @@ func (r *ServiceClaimReconciler) processServiceClaim(
 		StringData: map[string]string{},
 	}
 
+	var cel primazaiov1alpha1.ClusterEnvironmentList
+	if err := r.List(ctx, &cel); err != nil {
+		l.Info("error fetching ClusterEnvironmentList", "error", err)
+		return client.IgnoreNotFound(err)
+	}
+
 	// count the number of secret data entries
 	count := 0
 	// loop over every RegisteredService
 	registeredServiceFound := false
 	var registeredService primazaiov1alpha1.RegisteredService
-	for _, rs := range rsl.Items {
-		// Check if the ServiceClassIdentity given in ServiceClaim is a subset of
-		// ServiceClassIdentity given in the RegisteredService
-		if checkSCISubset(sclaim.Spec.ServiceClassIdentity, rs.Spec.ServiceClassIdentity) &&
-			envtag.Match(sclaim.Spec.EnvironmentTag, rs.Spec.Constraints.Environments) {
-			registeredServiceFound = true
-			registeredService = rs
-
-			var err error
-			count, err = r.extractServiceEndpointDefinition(ctx, req, rs, sclaim.Spec.ServiceEndpointDefinitionKeys, secret)
-			if err != nil {
-				l.Error(err, "unable to extract SED")
-				return err
+	if sclaim.Spec.EnvironmentTag == "" {
+		if sclaim.Spec.ApplicationClusterContext != nil {
+			for _, rs := range rsl.Items {
+				// Check if the ServiceClassIdentity given in ServiceClaim is a subset of
+				// ServiceClassIdentity given in the RegisteredService
+				if checkSCISubset(sclaim.Spec.ServiceClassIdentity, rs.Spec.ServiceClassIdentity) {
+					registeredServiceFound = true
+					registeredService = rs
+
+					var err error
+					count, err = r.extractServiceEndpointDefinition(ctx, req, rs, sclaim.Spec.ServiceEndpointDefinitionKeys, secret)
+					if err != nil {
+						l.Error(err, "unable to extract SED")
+						return err
+					}
+					break
+				}
+			}
+
+		}
+	} else {
+		for _, rs := range rsl.Items {
+			// Check if the ServiceClassIdentity given in ServiceClaim is a subset of
+			// ServiceClassIdentity given in the RegisteredService
+			if checkSCISubset(sclaim.Spec.ServiceClassIdentity, rs.Spec.ServiceClassIdentity) &&
+				envtag.Match(sclaim.Spec.EnvironmentTag, rs.Spec.Constraints.Environments) {
+				registeredServiceFound = true
+				registeredService = rs
+
+				var err error
+				count, err = r.extractServiceEndpointDefinition(ctx, req, rs, sclaim.Spec.ServiceEndpointDefinitionKeys, secret)
+				if err != nil {
+					l.Error(err, "unable to extract SED")
+					return err
+				}
+				break
```

Comment:
If the ApplicationClusterContext is defined, we should get the environment tag from `ClusterEnvironment` with name `ApplicationClusterContext.ClusterEnvironmentName`.

To me we should have something similar to this

```suggestion
		env := sclaim.Spec.EnvironmentTag
		if sclaim.Spec.ApplicationClusterContext != nil {
				env = r.getEnvironmentFromClusterEnvironment(ctx, sclaim.Spec.ApplicationClusterContext.ClusterEnvironmentName)
		}
		
		for _, rs := range rsl.Items {
			// Check if the ServiceClassIdentity given in ServiceClaim is a subset of
			// ServiceClassIdentity given in the RegisteredService
			if checkSCISubset(sclaim.Spec.ServiceClassIdentity, rs.Spec.ServiceClassIdentity) &&
				envtag.Match(sclaim.Spec.EnvironmentTag, env) {
				registeredServiceFound = true
				registeredService = rs

				var err error
				count, err = r.extractServiceEndpointDefinition(ctx, req, rs, sclaim.Spec.ServiceEndpointDefinitionKeys, secret)
				if err != nil {
					l.Error(err, "unable to extract SED")
					return err
				}
				break
```